### ISSUE TITLE
Some glob patterns unsupported for Exclude

### ISSUE THREAD

**Issue by mikeyhew** [rubocop-hq/rubocop#6105]
Some patterns that are supported by `Dir.glob`, like `bin/{rails,rake,yarn}`, don't work with the `Exclude` config in .rubycop.yaml. 

If this is a desired feature, and I'm guessing it is, I would be happy to try to implement it. Any tips would be appreciated :)

--------

## Expected behavior

`bin/rails`, `bin/rake`, and `bin/yarn` should be excluded from linting by rubucop

## Actual behavior

`bin/rails`, `bin/rake`, and `bin/yarn` are linted by rubocop

## Steps to reproduce the problem

Create files `bin/rails`, `bin/rake`, and `bin/yarn`, with ruby code in them that has some style issues that will be caught by rubocop lints. Make sure they have `#!/usr/bin/env ruby` at the top of them, so rubocop will treat them as ruby files. Then add this to `.rubocop.yml`:

```yaml
AllCops:
  Exclude:
    - 'bin/{rails,rake,yarn}'
```

and run `rubocop`.

## RuboCop version

0.58.1

**Comment by jonas054** [comment]
I agree that it would be good to add this kind of glob matching to `Inlude`, `Exclude` and any other parameters whose matching is implemented in `path_util.rb`. So take a look at `PathUtil.match_path?`. I think it's just a question of sending the right flags to `File.fnmatch?`. Good luck!
